src/amo/pages/UserProfileEdit/index.js

```diff
@@ -198,7 +200,13 @@ export class UserProfileEditBase extends React.Component<Props, State> {
     }
 
     if (wasUpdating && !isUpdating && !errorHandler.hasError()) {
```

Comment:
Ugh, checking for a change in the `isUpdating` prop is fragile because any return statement higher up in this function (I already see one) could result in the code never getting called. In any case, it was there before your patch so I guess it works but this is something we should avoid.

Reply:
The only `return` I see is for `!currentUser`, which seems valid that we wouldn't want/need to redirect in that case, but you are also correct that all I did was change some existing code. It does seem less than idea, but I'm not sure if we should "fix" it now or not.

Reply:
No, I wasn't suggesting to fix it now. It's something we should avoid in future patches, though.